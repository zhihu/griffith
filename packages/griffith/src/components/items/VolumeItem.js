import React from 'react'
import {css} from 'aphrodite/no-important'
import VolumeSlider from '../VolumeSlider'
import Icon from '../Icon'
import * as icons from '../icons/controller'
import styles from '../Controller.styles'

const VolumeItem = ({
  volume,
  menuShown,
  onMouseEnter,
  onMouseLeave,
  onDragStart,
  onDragEnd,
  onChange,
  onToggleMuted,
}) => (
  <div
    className={css(styles.menuContainer)}
    onMouseEnter={onMouseEnter}
    onMouseLeave={onMouseLeave}
  >
    <button className={css(styles.button)} onClick={onToggleMuted}>
      <Icon icon={volume === 0 ? icons.muted : icons.volume} />
    </button>
    <div className={css(styles.menu, menuShown && styles.menuShown)}>
      <VolumeSlider
        value={volume}
        total={1}
        onDragStart={onDragStart}
        onDragEnd={onDragEnd}
        onChange={onChange}
      />
    </div>
  </div>
)

export default VolumeItem
