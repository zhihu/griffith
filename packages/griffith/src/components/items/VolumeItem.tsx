import React from 'react'
import {css} from 'aphrodite/no-important'
import VolumeSlider, {VolumeSliderProps} from '../VolumeSlider'
import * as icons from '../icons/controller'
import styles from '../Controller.styles'
import ControllerButton from './ControllerButton'
import {useLocaleText} from '../../contexts/LocaleContext'

type VolumeItemProps = Pick<
  React.HTMLProps<HTMLDivElement>,
  'onMouseEnter' | 'onMouseLeave'
> &
  Pick<VolumeSliderProps, 'onChange' | 'onDragStart' | 'onDragEnd'> & {
    volume: number
    menuShown?: boolean
    onToggleMuted?(): void
  }

const VolumeItem = ({
  volume,
  menuShown,
  onMouseEnter,
  onMouseLeave,
  onDragStart,
  onDragEnd,
  onChange,
  onToggleMuted,
}: VolumeItemProps) => (
  <div
    className={css(styles.menuContainer)}
    onMouseEnter={onMouseEnter}
    onMouseLeave={onMouseLeave}
  >
    <ControllerButton
      aria-label={useLocaleText(volume === 0 ? 'unmute' : 'mute')}
      icon={volume === 0 ? icons.muted : icons.volume}
      onClick={() => onToggleMuted?.()}
    />
    <div className={css(styles.menu, menuShown && styles.menuShown)}>
      <VolumeSlider
        value={volume}
        total={1}
        onDragStart={onDragStart}
        onDragEnd={onDragEnd}
        onChange={onChange}
      />
    </div>
  </div>
)

export default VolumeItem
