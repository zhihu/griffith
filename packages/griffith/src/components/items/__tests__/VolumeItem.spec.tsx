import React from 'react'
import {shallow} from 'enzyme'
import VolumeItem from '../VolumeItem'

describe('VolumeItem', () => {
  it('get VolumeItem component', () => {
    const handleMouseEnter = jest.fn()
    const handlemouseLeave = jest.fn()
    const wrapper = shallow(
      <VolumeItem
        volume={0.9}
        menuShown={true}
        onMouseEnter={handleMouseEnter}
        onMouseLeave={handlemouseLeave}
        onToggleMuted={jest.fn()}
        onDragStart={jest.fn()}
        onDragEnd={jest.fn()}
        onChange={jest.fn()}
      />
    )
    expect(wrapper).toMatchSnapshot()

    // PointerEnter
    wrapper.simulate('mouseenter')
    expect(handleMouseEnter).toBeCalled()

    // PointerLeave
    wrapper.simulate('mouseleave')
    expect(handlemouseLeave).toBeCalled()
  })
})
