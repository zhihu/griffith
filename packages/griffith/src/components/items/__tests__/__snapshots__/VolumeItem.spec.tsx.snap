// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VolumeItem get VolumeItem component 1`] = `
<div>
  <div class="menuContainer_e296pg">
    <button aria-label="Mute" class="button_1j0k413" type="button">
      <span class="root_s9vkyf"
        ><svg viewBox="0 0 24 24" class="svg_11u3rwe">
          <path
            d="M5,12.0119396 L5,14.5238791 L8.33098415,14.5238791 L8.33098415,14.5238791 C8.39017333,14.520451 8.4497727,14.5223174 8.50904598,14.5296109 C8.72475585,14.5524598 8.91816198,14.6424266 9.06920843,14.7779841 L13,17.4293365 L13,12.0119396 L13,6.59454265 L9.06920843,9.24589504 C8.91816198,9.38145255 8.72475585,9.47141933 8.50904598,9.49426822 C8.4497727,9.50156175 8.39017333,9.5034281 8.33098415,9.5 L8.33098415,9.5 L5,9.5 L5,12.0119396 Z M3,12.0119396 L3,8.50157356 C3,8.2247344 3.11117675,7.97412842 3.29196292,7.79283868 C3.46994918,7.61192872 3.71994909,7.5 4.00000001,7.5 L8.08102098,7.5 L13.3535806,3.94361361 C13.5372091,3.81975463 13.7472793,3.76538554 13.9529683,3.77494994 C13.9685502,3.77423867 13.98423,3.77387913 14,3.77387913 C14.5522847,3.77387913 15,4.21934183 15,4.77259008 L15,12.0119396 L15,19.251289 C15,19.8045373 14.5522847,20.25 14,20.25 C13.98423,20.25 13.9685502,20.2496405 13.9529683,20.2489292 C13.7472793,20.2584936 13.5372091,20.2041245 13.3535806,20.0802655 L8.08102098,16.5238791 L4.00000001,16.5238791 C3.71994909,16.5238791 3.46994918,16.4119504 3.29196292,16.2310404 C3.11117675,16.0497507 3,15.7991447 3,15.5223056 L3,12.0119396 Z M17.5059844,15.7803571 L17.5,15.7726628 C18.6746368,14.8577452 19.4301566,13.4299638 19.4301566,11.8257212 C19.4301566,10.2214787 18.6746368,8.79369726 17.5,7.87877967 L17.5067574,7.87009164 C17.2041341,7.69816923 17,7.37292366 17,7 C17,6.44771525 17.4477153,6 18,6 C18.2836948,6 18.5397977,6.11813499 18.721799,6.30789526 L18.7279374,6.30000305 C20.3724289,7.58088767 21.4301566,9.57978168 21.4301566,11.8257212 C21.4301566,14.0716608 20.3724289,16.0705548 18.7279374,17.3514394 L18.7211832,17.3427554 C18.5392202,17.5321404 18.283375,17.6500092 18,17.6500092 C17.4477153,17.6500092 17,17.2022939 17,16.6500092 C17,16.2774031 17.2037866,15.9523939 17.5059844,15.7803571 L17.5059844,15.7803571 Z"
          ></path></svg
      ></span>
    </button>
    <div class="menu_1a7s4a3-o_O-menuShown_10w91hr">
      <div class="root_1cc9rz">
        <div
          class="root_17bkmwy-o_O-root_1bq0ydu-o_O-root_av3p6z"
          tabindex="0"
          role="slider"
        >
          <div class="inner_9gm5i8-o_O-inner_1jtvojh">
            <div class="track_1bopucd-o_O-track_31rx9n-o_O-track_ytl6g2">
              <div
                class="bar_1kpj1ed-o_O-bar_1ql2ook"
                style="
                  height: 100%;
                  transform: scaleY(0.9);
                  transform-origin: bottom;
                "
              ></div>
            </div>
            <div
              class="thumbWrapper_ay4wjb-o_O-thumbWrapper_drkmbm"
              style="
                height: 100%;
                transform: translateY(9.999999999999998%);
                transform-origin: bottom;
              "
            >
              <div class="thumb_ozdqky-o_O-thumb_8gf5gg"></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>

`;
